### PR TITLE
Allow restarting failed workflows

### PR DESCRIPTION
**Pull request by apmasell**
This adds an attempt number that a client can use to restart a failed workflow.
It also restart failed workflows if a new version of the workflow is available.

To accomplish this, there is an attempt number and if the workflow is failed
and the client sends a higher attempt number, the steps in the workflow will be
restarted (after updating the arguments).

### CODE REVIEW

**Review by mlaszloffy — COMMENTED** [review]
Can the api docs be updated to include "attempt"?

**Comment by apmasell** [comment]
Updated the Swagger.

File: vidarr-core/src/main/java/ca/on/oicr/gsi/vidarr/core/BaseProcessor.java [hunk 1]
```diff
@@ -89,11 +90,15 @@ public final synchronized void permanentFailure(String reason) {
...
-        try {
-          task.run();
-        } catch (Throwable e) {
-          System.err.println("Task thew exception. Failing workflow.");
-          permanentFailure(e.toString());
+        if (operation.isLive()) {
+          try {
+            task.run();
+          } catch (Throwable e) {
+            System.err.println("Task thew exception. Failing workflow.");
```
Comment (avarsava) [inline vidarr-core/src/main/java/ca/on/oicr/gsi/vidarr/core/BaseProcessor.java:97]: 'thew' -> 'threw' ?

**Review by avarsava — APPROVED** [review]

**Review by mlaszloffy — APPROVED** [review]

**Review by djcooke — APPROVED** [review]
`DatabaseBackedProcessor#submit` would be much easier to understand if it were refactored into smaller chunks. Moving each query into a separate method would help a lot.

Same goes for other methods that are hundreds of lines long and involve JOOQ queries, but this one was the most difficult to get through

**Pull request merged**